### ISSUE TITLE
SafariDriver doesn't work with Selenium 2.46.0.

### ISSUE THREAD

**Issue by Jamizzle93** [SeleniumHQ/selenium#631]
I believe to have ran into an issue after updating to Selenium 2.46.

When running Safari tests using SafariDriver with Selenium 2.46 on a node connected to my grid, I receive the following error.

```exec error: Error: Command failed: com.google.gson.JsonObject cannot be cast to java.lang.String (java.lang.ClassCastException) (Selenium::WebDriver::Error::UnknownError)
[remote server] org.openqa.selenium.remote.JsonToBeanConverter(JsonToBeanConverter.java):111:in `convert'
[remote server] org.openqa.selenium.remote.JsonToBeanConverter(JsonToBeanConverter.java):42:in `convert'
[remote server] org.openqa.selenium.safari.SafariDriverCommandExecutor(SafariDriverCommandExecutor.java):191:in `execute'
[remote server] org.openqa.selenium.remote.RemoteWebDriver(RemoteWebDriver.java):578:in `execute'
[remote server] org.openqa.selenium.remote.RemoteWebDriver(RemoteWebDriver.java):242:in `startSession'
[remote server] org.openqa.selenium.remote.RemoteWebDriver(RemoteWebDriver.java):128:in `<init>'
[remote server] org.openqa.selenium.remote.RemoteWebDriver(RemoteWebDriver.java):141:in `<init>'
[remote server] org.openqa.selenium.safari.SafariDriver(SafariDriver.java):58:in `<init>'
[remote server] org.openqa.selenium.safari.SafariDriver(SafariDriver.java):51:in `<init>'
[remote server] sun.reflect.NativeConstructorAccessorImpl(NativeConstructorAccessorImpl.java):-2:in `newInstance0'
[remote server] sun.reflect.NativeConstructorAccessorImpl(NativeConstructorAccessorImpl.java):62:in `newInstance'
[remote server] sun.reflect.DelegatingConstructorAccessorImpl(DelegatingConstructorAccessorImpl.java):45:in `newInstance'
[remote server] java.lang.reflect.Constructor(Constructor.java):408:in `newInstance'
[remote server] org.openqa.selenium.remote.server.DefaultDriverProvider(DefaultDriverProvider.java):85:in `callConstructor'
[remote server] org.openqa.selenium.remote.server.DefaultDriverProvider(DefaultDriverProvider.java):79:in `newInstance'
[remote server] org.openqa.selenium.remote.server.DefaultDriverFactory(DefaultDriverFactory.java):62:in `newInstance'
[remote server] org.openqa.selenium.remote.server.DefaultSession$BrowserCreator(DefaultSession.java):222:in `call'
[remote server] org.openqa.selenium.remote.server.DefaultSession$BrowserCreator(DefaultSession.java):1:in `call'
[remote server] java.util.concurrent.FutureTask(FutureTask.java):266:in `run'
[remote server] org.openqa.selenium.remote.server.DefaultSession$1(DefaultSession.java):176:in `run'
[remote server] java.util.concurrent.ThreadPoolExecutor(ThreadPoolExecutor.java):1142:in `runWorker'
[remote server] java.util.concurrent.ThreadPoolExecutor$Worker(ThreadPoolExecutor.java):617:in `run'
[remote server] java.lang.Thread(Thread.java):745:in `run'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/response.rb:71:in `assert_ok'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/response.rb:34:in `initialize'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/http/common.rb:78:in `new'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/http/common.rb:78:in `create_response'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/http/default.rb:90:in `request'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/http/common.rb:59:in `call'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/bridge.rb:657:in `raw_execute'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/bridge.rb:122:in `create_session'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/remote/bridge.rb:87:in `initialize'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/common/driver.rb:52:in `new'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver/common/driver.rb:52:in `for'
/var/lib/gems/1.9.1/gems/selenium-webdriver-2.46.2/lib/selenium/webdriver.rb:84:in `for'
/vagrant/tests/cucumber/features/support/env.rb:67:in `<top (required)>'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/rb_support/rb_language.rb:95:in `load'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/rb_support/rb_language.rb:95:in `load_code_file'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/runtime/support_code.rb:180:in `load_file'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/runtime/support_code.rb:83:in `block in load_files!'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/runtime/support_code.rb:82:in `each'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/runtime/support_code.rb:82:in `load_files!'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/runtime.rb:184:in `load_step_definitions'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/runtime.rb:42:in `run!'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/lib/cucumber/cli/main.rb:47:in `execute!'
/var/lib/gems/1.9.1/gems/cucumber-1.3.19/bin/cucumber:13:in `<top (required)>'
/usr/local/bin/cucumber:23:in `load'
/usr/local/bin/cucumber:23:in `<main>'```

I have built the SafariDriver by pulling down the latest code from the Selenium repository and building the driver that way.

Upgrading to 2.46 fixed a Firefox 38 not running on 2.45, but it's damaged Safari unfortunately.

**Comment by bgd77** [comment]
Having the same error when using SafariDriver 2.45 and Selenium 2.46. There is no SafariDriver 2.46 yet.

org.openqa.selenium.WebDriverException: com.google.gson.JsonObject cannot be cast to java.lang.String
Command duration or timeout: 2.64 seconds
Build info: version: '2.45.0', revision: '5017cb8e7ca8e37638dc3091b2440b90a1d8686f', time: '2015-02-27 09:10:26'
System info: host: , os.name: 'Linux', os.arch: 'amd64', os.version: '2.6.32-431.20.3.el6.x86_64', java.version: '1.7.0_55'
Driver info: org.openqa.selenium.remote.RemoteWebDriver
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_55]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57) ~[na:1.7.0_55]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.7.0_55]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526) ~[na:1.7.0_55]
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:204) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:156) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:599) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:240) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:126) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:153) ~[selenium-remote-driver-2.45.0.jar:na]
.
.
.
Caused by: java.lang.ClassCastException: com.google.gson.JsonObject cannot be cast to java.lang.String
	at org.openqa.selenium.remote.JsonToBeanConverter.convert(JsonToBeanConverter.java:111) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.JsonToBeanConverter.convert(JsonToBeanConverter.java:42) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.safari.SafariDriverCommandExecutor.execute(SafariDriverCommandExecutor.java:191) ~[selenium-safari-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:578) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:242) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:128) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:141) ~[selenium-remote-driver-2.45.0.jar:na]
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:58) ~[selenium-safari-driver-2.45.0.jar:na]
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:51) ~[selenium-safari-driver-2.45.0.jar:na]

**Comment by barancev** [comment]
Fixed by commit 26bf0bc54be8cccab99f4fb2b170bb9aa8ad8030

**Comment by perwez123** [comment]
I too have faced the same issue when my test were running in node , The browsers opened and closed immediately . My test were running nicely on selenium 2.45.0 version but this issue occurred since  I updated selenium jars to 2.46.0 .  I had not installed safari driver in node system , i used only extension file that was available on 2.45.0 version .

When can we get this fixes , Can you please provide some  steps to install the safari driver in node system  which comes with 2.46.0 version 

It thrown the following error in node system.


18:14:23.641 INFO - Driver class not found: com.opera.core.systems.OperaDriver
18:14:23.641 INFO - Driver provider com.opera.core.systems.OperaDriver is not registered
18:14:23.674 INFO - Selenium Grid node is up and ready to register to the hub
18:14:23.692 INFO - Starting auto registration thread. Will try to register every 5000 ms.
18:14:23.692 INFO - Registering the node to the hub: http://10.98.1.30:4444/grid/register
18:14:23.704 INFO - The node is registered to the hub and ready to use
18:14:44.669 INFO - Executing: [new session: Capabilities [{platform=ANY, browserName=safari, version=, safari.options={port=0, cleanSession=true}}]])
18:14:44.675 INFO - Creating a new session for Capabilities [{platform=ANY, browserName=safari, version=, safari.options={port=0, cleanSession=true}}]
18:14:44.728 INFO - Server started on port 47355
18:14:44.736 INFO - Launching Safari
18:14:44.744 INFO - Waiting for SafariDriver to connect
18:14:46.106 INFO - Connection opened
18:14:46.107 INFO - Driver connected in 1363 ms
18:14:46.117 INFO - Shutting down
18:14:46.117 INFO - Closing connection
18:14:46.118 INFO - Stopping Safari
18:14:46.171 INFO - Stopping server
18:14:46.171 INFO - Stopping server
18:14:46.175 INFO - Shutdown complete
18:14:46.400 WARN - Exception thrown
java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'CHENNMW-0YVF8J5.local', ip: '10.98.5.239', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.9.5', java.version: '1.7.0_71'
Driver info: driver.version: unknown
at java.util.concurrent.FutureTask.report(FutureTask.java:122)
at java.util.concurrent.FutureTask.get(FutureTask.java:188)
at org.openqa.selenium.remote.server.DefaultSession.execute(DefaultSession.java:183)
at org.openqa.selenium.remote.server.DefaultSession.(DefaultSession.java:119)
at org.openqa.selenium.remote.server.DefaultSession.createSession(DefaultSession.java:95)
at org.openqa.selenium.remote.server.DefaultDriverSessions.newSession(DefaultDriverSessions.java:137)
at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:58)
at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:1)
at org.openqa.selenium.remote.server.rest.ResultConfig.handle(ResultConfig.java:111)
at org.openqa.selenium.remote.server.JsonHttpCommandHandler.handleRequest(JsonHttpCommandHandler.java:174)
at org.openqa.selenium.remote.server.DriverServlet.handleRequest(DriverServlet.java:202)
at org.openqa.selenium.remote.server.DriverServlet.doPost(DriverServlet.java:164)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:727)
at org.openqa.selenium.remote.server.DriverServlet.service(DriverServlet.java:130)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
at org.openqa.jetty.jetty.servlet.ServletHolder.handle(ServletHolder.java:428)
at org.openqa.jetty.jetty.servlet.ServletHandler.dispatch(ServletHandler.java:680)
at org.openqa.jetty.jetty.servlet.ServletHandler.handle(ServletHandler.java:571)
at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1526)
at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1479)
at org.openqa.jetty.http.HttpServer.service(HttpServer.java:920)
at org.openqa.jetty.http.HttpConnection.service(HttpConnection.java:820)
at org.openqa.jetty.http.HttpConnection.handleNext(HttpConnection.java:986)
at org.openqa.jetty.http.HttpConnection.handle(HttpConnection.java:837)
at org.openqa.jetty.http.SocketListener.handleConnection(SocketListener.java:243)
at org.openqa.jetty.util.ThreadedServer.handle(ThreadedServer.java:358)
at org.openqa.jetty.util.ThreadPool$PoolThread.run(ThreadPool.java:537)
Caused by: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'CHENNMW-0YVF8J5.local', ip: '10.98.5.239', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.9.5', java.version: '1.7.0_71'
Driver info: driver.version: unknown
at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor(DefaultDriverProvider.java:95)
at org.openqa.selenium.remote.server.DefaultDriverProvider.newInstance(DefaultDriverProvider.java:79)
at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance(DefaultDriverFactory.java:62)
at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:222)
at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:1)
at java.util.concurrent.FutureTask.run(FutureTask.java:262)
at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor(DefaultDriverProvider.java:85)
... 9 more
Caused by: org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'CHENNMW-0YVF8J5.local', ip: '10.98.5.239', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.9.5', java.version: '1.7.0_71'
Driver info: driver.version: SafariDriver
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:599)
at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:242)
at org.openqa.selenium.remote.RemoteWebDriver.(RemoteWebDriver.java:128)
at org.openqa.selenium.remote.RemoteWebDriver.(RemoteWebDriver.java:141)
at org.openqa.selenium.safari.SafariDriver.(SafariDriver.java:58)
at org.openqa.selenium.safari.SafariDriver.(SafariDriver.java:51)
... 14 more
Caused by: java.lang.ClassCastException: com.google.gson.JsonObject cannot be cast to java.lang.String
at org.openqa.selenium.remote.JsonToBeanConverter.convert(JsonToBeanConverter.java:111)
at org.openqa.selenium.remote.JsonToBeanConverter.convert(JsonToBeanConverter.java:42)
at org.openqa.selenium.safari.SafariDriverCommandExecutor.execute(SafariDriverCommandExecutor.java:191)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:578)
... 19 more
18:14:46.404 WARN - Exception: com.google.gson.JsonObject cannot be cast to java.lang.String

**Comment by sayaara** [comment]
I am facing same issue. Safari browser gets closed immediately after launching the browser with java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException

Selenium 2.46.
Mac OS X 10.10.3 x86_64
Safari 8.0.5
---------------------------------------
Here is my sample python code

from selenium import webdriver

def test():
    driver = webdriver.Safari()
    driver.get("www.google.com")

when I run the above function, I am getting the following output
---------------------------------------------------
15:57:46.465 INFO - Launching a standalone Selenium Server
15:57:46.491 INFO - Java: Oracle Corporation 24.45-b08
15:57:46.491 INFO - OS: Mac OS X 10.10.3 x86_64
15:57:46.500 INFO - v2.46.0, with Core v2.46.0. Built from revision 87c69e2
15:57:46.551 INFO - Driver provider org.openqa.selenium.ie.InternetExplorerDriver registration is skipped:
registration capabilities Capabilities [{platform=WINDOWS, ensureCleanSession=true, browserName=internet explorer, version=}] does not match the current platform MAC
15:57:46.552 INFO - Driver class not found: com.opera.core.systems.OperaDriver
15:57:46.552 INFO - Driver provider com.opera.core.systems.OperaDriver is not registered
15:57:46.603 INFO - RemoteWebDriver instances should connect to: http://127.0.0.1:50220/wd/hub
15:57:46.603 INFO - Selenium Server is up and running
15:57:56.500 INFO - Executing: [new session: Capabilities [{platform=ANY, javascriptEnabled=true, browserName=safari, version=}]])
15:57:56.508 INFO - Creating a new session for Capabilities [{platform=ANY, javascriptEnabled=true, browserName=safari, version=}]
15:57:56.561 INFO - Server started on port 39602
15:57:56.570 INFO - Launching Safari
15:57:56.579 INFO - Waiting for SafariDriver to connect
15:57:58.044 INFO - Connection opened
15:57:58.046 INFO - Driver connected in 1466 ms
15:57:58.071 INFO - Shutting down
15:57:58.071 INFO - Closing connection
15:57:58.072 INFO - Stopping Safari
15:57:58.123 INFO - Stopping server
15:57:58.124 INFO - Stopping server
15:57:58.132 INFO - Shutdown complete
15:57:58.134 WARN - Exception thrown
java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'L502062.local', ip: '10.10.80.7', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.10.3', java.version: '1.7.0_45'
Driver info: driver.version: unknown
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:188)
	at org.openqa.selenium.remote.server.DefaultSession.execute(DefaultSession.java:183)
	at org.openqa.selenium.remote.server.DefaultSession.<init>(DefaultSession.java:119)
	at org.openqa.selenium.remote.server.DefaultSession.createSession(DefaultSession.java:95)
	at org.openqa.selenium.remote.server.DefaultDriverSessions.newSession(DefaultDriverSessions.java:137)
	at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:58)
	at org.openqa.selenium.remote.server.handler.NewSession.handle(NewSession.java:1)
	at org.openqa.selenium.remote.server.rest.ResultConfig.handle(ResultConfig.java:111)
	at org.openqa.selenium.remote.server.JsonHttpCommandHandler.handleRequest(JsonHttpCommandHandler.java:174)
	at org.openqa.selenium.remote.server.DriverServlet.handleRequest(DriverServlet.java:202)
	at org.openqa.selenium.remote.server.DriverServlet.doPost(DriverServlet.java:164)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:727)
	at org.openqa.selenium.remote.server.DriverServlet.service(DriverServlet.java:130)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
	at org.openqa.jetty.jetty.servlet.ServletHolder.handle(ServletHolder.java:428)
	at org.openqa.jetty.jetty.servlet.ServletHandler.dispatch(ServletHandler.java:680)
	at org.openqa.jetty.jetty.servlet.ServletHandler.handle(ServletHandler.java:571)
	at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1526)
	at org.openqa.jetty.http.HttpContext.handle(HttpContext.java:1479)
	at org.openqa.jetty.http.HttpServer.service(HttpServer.java:920)
	at org.openqa.jetty.http.HttpConnection.service(HttpConnection.java:820)
	at org.openqa.jetty.http.HttpConnection.handleNext(HttpConnection.java:986)
	at org.openqa.jetty.http.HttpConnection.handle(HttpConnection.java:837)
	at org.openqa.jetty.http.SocketListener.handleConnection(SocketListener.java:243)
	at org.openqa.jetty.util.ThreadedServer.handle(ThreadedServer.java:358)
	at org.openqa.jetty.util.ThreadPool$PoolThread.run(ThreadPool.java:537)
Caused by: org.openqa.selenium.WebDriverException: java.lang.reflect.InvocationTargetException
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'L502062.local', ip: '10.10.80.7', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.10.3', java.version: '1.7.0_45'
Driver info: driver.version: unknown
	at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor(DefaultDriverProvider.java:95)
	at org.openqa.selenium.remote.server.DefaultDriverProvider.newInstance(DefaultDriverProvider.java:79)
	at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance(DefaultDriverFactory.java:62)
	at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:222)
	at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call(DefaultSession.java:1)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:176)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:744)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor(DefaultDriverProvider.java:85)
	... 9 more
Caused by: org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
System info: host: 'L502062.local', ip: '10.10.80.7', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.10.3', java.version: '1.7.0_45'
Driver info: driver.version: SafariDriver
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:599)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:242)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:128)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:141)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:58)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:51)
	... 14 more
Caused by: java.lang.ClassCastException: com.google.gson.JsonObject cannot be cast to java.lang.String
	at org.openqa.selenium.remote.JsonToBeanConverter.convert(JsonToBeanConverter.java:111)
	at org.openqa.selenium.remote.JsonToBeanConverter.convert(JsonToBeanConverter.java:42)
	at org.openqa.selenium.safari.SafariDriverCommandExecutor.execute(SafariDriverCommandExecutor.java:191)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:578)
	... 19 more
15:57:58.139 WARN - Exception: com.google.gson.JsonObject cannot be cast to java.lang.String
[ WARN ] Keyword 'Capture Page Screenshot' could not be run on failure: Cannot access execution context
Traceback (most recent call last):
  File "/Users/veeraa/Documents/workspace/med_selenium/lib/ui/safari_test.py", line 30, in <module>
    MyClass().test()
  File "<string>", line 2, in test
  File "/Library/Python/2.7/site-packages/Selenium2Library/keywords/keywordgroup.py", line 15, in _run_on_failure_decorator
    return method(*args, **kwargs)
  File "/Users/veeraa/Documents/workspace/med_selenium/lib/ui/safari_test.py", line 26, in test
    self.driver = webdriver.Safari()
  File "/Library/Python/2.7/site-packages/selenium/webdriver/safari/webdriver.py", line 62, in __init__
    desired_capabilities=desired_capabilities)
  File "/Library/Python/2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 89, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File "/Library/Python/2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 138, in start_session
    'desiredCapabilities': desired_capabilities,
  File "/Library/Python/2.7/site-packages/Selenium2Library/webdrivermonkeypatches.py", line 11, in execute
    result = self._base_execute(driver_command, params)
  File "/Library/Python/2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 195, in execute
    self.error_handler.check_response(response)
  File "/Library/Python/2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 170, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: com.google.gson.JsonObject cannot be cast to java.lang.String
Stacktrace:
    at org.openqa.selenium.remote.JsonToBeanConverter.convert (JsonToBeanConverter.java:111)
    at org.openqa.selenium.remote.JsonToBeanConverter.convert (JsonToBeanConverter.java:42)
    at org.openqa.selenium.safari.SafariDriverCommandExecutor.execute (SafariDriverCommandExecutor.java:191)
    at org.openqa.selenium.remote.RemoteWebDriver.execute (RemoteWebDriver.java:578)
    at org.openqa.selenium.remote.RemoteWebDriver.startSession (RemoteWebDriver.java:242)
    at org.openqa.selenium.remote.RemoteWebDriver.<init> (RemoteWebDriver.java:128)
    at org.openqa.selenium.remote.RemoteWebDriver.<init> (RemoteWebDriver.java:141)
    at org.openqa.selenium.safari.SafariDriver.<init> (SafariDriver.java:58)
    at org.openqa.selenium.safari.SafariDriver.<init> (SafariDriver.java:51)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0 (NativeConstructorAccessorImpl.java:-2)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:526)
    at org.openqa.selenium.remote.server.DefaultDriverProvider.callConstructor (DefaultDriverProvider.java:85)
    at org.openqa.selenium.remote.server.DefaultDriverProvider.newInstance (DefaultDriverProvider.java:79)
    at org.openqa.selenium.remote.server.DefaultDriverFactory.newInstance (DefaultDriverFactory.java:62)
    at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call (DefaultSession.java:222)
    at org.openqa.selenium.remote.server.DefaultSession$BrowserCreator.call (DefaultSession.java:1)
    at java.util.concurrent.FutureTask.run (FutureTask.java:262)
    at org.openqa.selenium.remote.server.DefaultSession$1.run (DefaultSession.java:176)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:615)
    at java.lang.Thread.run (Thread.java:744)


Appreciate if anyone can help me to resolve this issue?

Thanks much

**Comment by nmn** [comment]
Same issue here. In order to not pollute yet, I'm holding back my error logs. Which look similar to the ones above.

**Comment by ddavison** [comment]
as barancev had said - it was fixed in commit 26bf0bc54be8cccab99f4fb2b170bb9aa8ad8030 . you can build your own client libraries, or wait until a new version of the API is released with this fix

Locking.
